src/components/Synthetics/Claims/Claims.tsx

```diff
@@ -13,98 +13,161 @@ import "./Claims.scss";
-  });
+  const [isAccruedPositionPriceImpactRebateModalVisible, setIsAccruedPositionPriceImpactRebateModalVisible] =
+    useState(false);
+  const [isClaimablePositionPriceImpactFeesModalVisible, setIsClaimablePositionPriceImpactFeesModalVisible] =
+    useState(false);
 
-  const isEmpty = !account || claimActions?.length === 0;
+  const { claimActions, isLoading, accruedPositionPriceImpactFees, claimablePositionPriceImpactFees } =
+    useClaimCollateralHistory(chainId, {
+      marketsInfoData,
+      tokensData,
+      pageIndex,
+      pageSize: PAGE_SIZE,
+    });
 
-  const handleClaimClick = useCallback(() => {
-    setIsClaiming(true);
-  }, [setIsClaiming]);
+  const isEmpty = !account || claimActions?.length === 0;
 
-  const handleSettleClick = useCallback(() => {
-    setIsSettling(true);
-  }, [setIsSettling]);
+  const handleClaimClick = useCallback(() => setIsClaiming(true), [setIsClaiming]);
+  const handleSettleClick = useCallback(() => setIsSettling(true), [setIsSettling]);
+  const handleAccruedPositionPriceImpactRebateClick = useCallback(
+    () => setIsAccruedPositionPriceImpactRebateModalVisible(true),
+    [setIsAccruedPositionPriceImpactRebateModalVisible]
+  );
+  const handleClaimablePositionPriceImpactFeesClick = useCallback(
+    () => setIsClaimablePositionPriceImpactFeesModalVisible(true),
+    [setIsClaimablePositionPriceImpactFeesModalVisible]
+  );
 
   const isMobile = useMedia("(max-width: 1100px)");
 
   return (
-    <div className="TradeHistory">
-      {account && isLoading && (
-        <div className="TradeHistoryRow App-box">
-          <Trans>Loading...</Trans>
-        </div>
-      )}
-      <div
-        className={cx("flex", "w-full", {
-          "flex-column": isMobile,
-        })}
-      >
-        {account && !isLoading && (
-          <SettleAccruedCard
-            positionsInfoData={positionsInfoData}
-            onSettleClick={handleSettleClick}
-            style={isMobile ? undefined : { marginRight: 4 }}
-          />
-        )}
-        {account && !isLoading && (
-          <ClaimableCard
-            marketsInfoData={marketsInfoData}
-            onClaimClick={handleClaimClick}
-            style={isMobile ? undefined : { marginLeft: 4 }}
-          />
+    <>
+      <ClaimModal
+        marketsInfoData={marketsInfoData}
+        isVisible={isClaiming}
+        onClose={() => setIsClaiming(false)}
+        setPendingTxns={setPendingTxns}
+      />
+      <SettleAccruedFundingFeeModal
+        isVisible={isSettling}
+        positionKeys={gettingPendingFeePositionKeys}
+        positionsInfoData={positionsInfoData}
+        tokensData={tokensData}
+        allowedSlippage={allowedSlippage}
+        setPositionKeys={setGettingPendingFeePositionKeys}
+        setPendingTxns={setPendingTxns}
+        onClose={useCallback(() => {
+          setGettingPendingFeePositionKeys([]);
+          setIsSettling(false);
+        }, [setGettingPendingFeePositionKeys, setIsSettling])}
```

Comment:
we can store the useCallback into a variable. Hard to see the use of a hook in the middle.

Reply:
fixed